05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,31 @@
+const jwt = require("jsonwebtoken");
+const { BadRequestError } = require("../errors");
+
+const login = async (req, res) => {
+  console.log("Login route handler called");
+  const { username, password } = req.body;
+  console.log("USERNAME:", username, "PASSWORD:", password);
+  if (!username || !password) {
+    throw new BadRequestError("please provide email and password");
+  }
+  const id = new Date().getDate();
+
+  const token = jwt.sign({ id, username }, process.env.JWT_SECRET, {
+    expiresIn: "30d",
+  });
+  res.status(200).json({ msg: "user created", token });
+  console.log(token);
```

Comment:
Nitpick: 
I'd recommend moving any `console.log(...)` statements above calling `res.status(...).json(...)`. You should think of this statement as the last thing that should be invoked since it flushes the response stream and returns a response.